single_number.py

```diff
@@ -0,0 +1,34 @@
+# Problem: Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
+# You must implement a solution with a linear runtime complexity and use only constant extra space.
+from collections import defaultdict
```

Comment:
Is this file new? Lots of things need to be fixed:

`camel_case_with_underscores`, not `camelCase`
- Docstrings, typehints,
- Solution can be made more efficient by just XORing all elements of list